python/lsst/daf/butler/registry/interfaces/_database.py

```diff
@@ -265,6 +265,31 @@ def dropTemporaryTable(self, table: sqlalchemy.schema.Table) -> None:
         else:
             raise TypeError(f"Table {table.key} was not created by makeTemporaryTable.")
 
+    @contextmanager
+    def temporary_table(self, spec: ddl.TableSpec, name: Optional[str]) -> Iterator[sqlalchemy.schema.Table]:
```

Comment:
I don't see this used anywhere, I guess it's also for one of the other branches?

Reply:
Correct.  An upcoming branch will use it.  I suppose I could also use it to simplify the existing `[Something]QueryResult.materialize` context managers, but I'll be replacing them completely in one of those upcoming branches, too.